llvm/lib/Transforms/InstCombine/InstCombineMulDivRem.cpp

```diff
@@ -1820,6 +1989,30 @@ Instruction *InstCombinerImpl::visitFDiv(BinaryOperator &I) {
     return R;
 
   Value *Op0 = I.getOperand(0), *Op1 = I.getOperand(1);
+
+  // Convert
+  // x = 1.0/sqrt(a)
+  // r1 = x * x;
+  // r2 = a * x;
+  //
+  // TO
+  //
+  // tmp1 = 1.0 / a
+  // tmp2 = sqrt (a)
+  // tmp3 = tmp1 * tmp2
+  // x = tmp3
+  // r1 = tmp1
+  // r2 = tmp2
```

Comment:
Rephrase the output comment in terms of the original value names

Reply:
Didn't get this. Could you please elaborate or tell me how would that look?

Reply:
should the output comment be:
// Transform ....
// ....
// TO
// r1 = 1/a
// r2 = sqrt(a)
// x = r1 * r2

Is  this what you are expecting? This wont give the correct picture of the transformation as one may think that we are hoisting the ops for r1/r2 above x which is incorrect.

Reply:
@yashssh As said above, this might give the reader an incorrect picture that we are hoisting the r1/r2 above x. We are just hoisting the operations corresponding to r1/r2 above x.
If you and @arsenm still suggest to modify this, will change it